# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidence to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?


In [1]:
import numpy as np
import pandas as pd

In [3]:
df = pd.read_csv(r'C:\Users\adoro\OneDrive\Desktop\CT Python\Week Three\Assignments\AB_NYC_2019.csv')
df


id                                               name   host_id  \
0          2539                 Clean & quiet apt home by the park      2787   
1          2595                              Skylit Midtown Castle      2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !      4632   
3          3831                    Cozy Entire Floor of Brownstone      4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park      7192   
...         ...                                                ...       ...   
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-10-19               0.21                               6   
1      2019-05-21               0.38                               2   
2             NaN                NaN                               1   
3      2019-07-05               4.64                               1   
4      2018-11-19               0.10                               1   
...           ...                ...                             ...   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 16 columns]

In [206]:
# Which hosts are the busiest and why?

df.groupby(['host_name']).count().sort_values(['availability_365'], ascending = False)[['availability_365', 'reviews_per_month','calculated_host_listings_count']]



availability_365  reviews_per_month  \
host_name                                           
Michael                    417                335   
David                      403                309   
Sonder (NYC)               327                207   
John                       294                250   
Alex                       279                229   
...                        ...                ...   
Jerbean                      1                  1   
Jerald                       1                  1   
Jeonghoon                    1                  0   
Jeny                         1                  1   
현선                           1                  0   

              calculated_host_listings_count  
host_name                                     
Michael                                  417  
David                                    403  
Sonder (NYC)                             327  
John                                     294  
Alex                                     279  
...                                      ...  
Jerbean                                    1  
Jerald                                     1  
Jeonghoon                                  1  
Jeny                                       1  
현선                                         1  

[11452 rows x 3 columns]

To determine which hosts were the busiest, we wanted to focus on the availability of the listings, the number of listings, and the number of reviews the hosts receive. Based on the information above, we believe that John is the busiest as he has the highest review per availability while having a top 5 amount of listings.

In [65]:
# How many neighborhood groups are available and which shows up the most?
df.groupby(['neighbourhood_group']).count().sort_values(['neighbourhood_group'], ascending = True)


id   name  host_id  host_name  neighbourhood  \
neighbourhood_group                                                    
Bronx                 1091   1090     1091       1090           1091   
Brooklyn             20104  20098    20104      20095          20104   
Manhattan            21661  21652    21661      21652          21661   
Queens                5666   5666     5666       5664           5666   
Staten Island          373    373      373        373            373   

                     latitude  longitude  room_type  price  minimum_nights  \
neighbourhood_group                                                          
Bronx                    1091       1091       1091   1091            1091   
Brooklyn                20104      20104      20104  20104           20104   
Manhattan               21661      21661      21661  21661           21661   
Queens                   5666       5666       5666   5666            5666   
Staten Island             373        373        373    373             373   

                     number_of_reviews  last_review  reviews_per_month  \
neighbourhood_group                                                      
Bronx                             1091          876                876   
Brooklyn                         20104        16447              16447   
Manhattan                        21661        16632              16632   
Queens                            5666         4574               4574   
Staten Island                      373          314                314   

                     calculated_host_listings_count  availability_365  
neighbourhood_group                                                    
Bronx                                          1091              1091  
Brooklyn                                      20104             20104  
Manhattan                                     21661             21661  
Queens                                         5666              5666  
Staten Island                                   373               373

Based on our review, there are 5 neighborhood groups within New York: Bronx, Brooklyn, Manhattan, Queens, and Staten Island. Manhattan was the one that showed up the most with 21,661 listings. The 21,661 listings were found under the id number as all listings will have an unique number.

In [68]:
# Are private rooms the most popular in manhattan?
df.groupby(['neighbourhood_group', 'room_type']).count().sort_values(['neighbourhood_group'])[['calculated_host_listings_count']]


calculated_host_listings_count
neighbourhood_group room_type                                      
Bronx               Entire home/apt                             379
                    Private room                                652
                    Shared room                                  60
Brooklyn            Entire home/apt                            9559
                    Private room                              10132
                    Shared room                                 413
Manhattan           Entire home/apt                           13199
                    Private room                               7982
                    Shared room                                 480
Queens              Entire home/apt                            2096
                    Private room                               3372
                    Shared room                                 198
Staten Island       Entire home/apt                             176
                    Private room                                188
                    Shared room                                   9

Based on our review, entire home/apt are the most popular in Manhattan with 13,199 listings.
Privates rooms were second in popularity in Manhattan with 7982 listings.

In [195]:
# Which hosts are the busiest and based on their reviews?

df.groupby(['host_name']).count().sort_values(['reviews_per_month'], ascending = False)[['calculated_host_listings_count', 'reviews_per_month', 'availability_365']]

calculated_host_listings_count  reviews_per_month  \
host_name                                                         
Michael                                  417                335   
David                                    403                309   
John                                     294                250   
Alex                                     279                229   
Sonder (NYC)                             327                207   
...                                      ...                ...   
Joscelyn                                   1                  0   
Dae                                        1                  0   
Navarra                                    1                  0   
Dagmara                                    1                  0   
현선                                         1                  0   

              availability_365  
host_name                       
Michael                    417  
David                      403  
John                       294  
Alex                       279  
Sonder (NYC)               327  
...                        ...  
Joscelyn                     1  
Dae                          1  
Navarra                      1  
Dagmara                      1  
현선                           1  

[11452 rows x 3 columns]

Similar to our review above, we believe John was the busiest per capita of reviews per month to calculated host listings. The second busiest was Alex as he was second in reviews per listings.

In [91]:
#Which neighorhood group has the highest average price?
df.groupby(['neighbourhood_group']).mean().sort_values(['price'], ascending = False)[['price']].round(2)

C:\Users\adoro\AppData\Local\Temp\ipykernel_13976\3352755084.py:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby(['neighbourhood_group']).mean().sort_values(['price'], ascending = False)[['price']].round(2)


price
neighbourhood_group        
Manhattan            196.88
Brooklyn             124.38
Staten Island        114.81
Queens                99.52
Bronx                 87.50

Based on our findings by filtering the data to show neighbourhood groups and average price, Manhattan had the highest average price of $196.88.

In [97]:
# Which neighbor hood group has the highest total price?
df.groupby(['neighbourhood_group']).max().sort_values(['price'], ascending = False)[['price']]

C:\Users\adoro\AppData\Local\Temp\ipykernel_13976\2803579284.py:2: FutureWarning: Dropping invalid columns in DataFrameGroupBy.max is deprecated. In a future version, a TypeError will be raised. Before calling .max, select only columns which should be valid for the function.
  df.groupby(['neighbourhood_group']).max().sort_values(['price'], ascending = False)[['price']]


price
neighbourhood_group       
Brooklyn             10000
Manhattan            10000
Queens               10000
Staten Island         5000
Bronx                 2500

In preparation for this question, we filtered the data by neighbourhood group and price and then selected the top 5 maximum amounts. Our analysis shows that Brooklyn, Manhattan, and Queens each charge a high price of $10,000.

In [98]:
#Which top 5 hosts have the highest total price?
df.groupby(['host_name']).max().sort_values(['price'], ascending = False)[['price']].head(5)

C:\Users\adoro\AppData\Local\Temp\ipykernel_13976\2702354918.py:2: FutureWarning: Dropping invalid columns in DataFrameGroupBy.max is deprecated. In a future version, a TypeError will be raised. Before calling .max, select only columns which should be valid for the function.
  df.groupby(['host_name']).max().sort_values(['price'], ascending = False)[['price']].head(5)


price
host_name       
Kathrine   10000
Erin       10000
Jelena     10000
Amy         9999
Olson       9999

After filtering the data by host name and price, we have found that Kathrine, Erin and Jelena charge a price of $10,000 and Amy/Olson charge a price of $9999.

In [116]:
# Who currently has no (zero) availability with a review count of 100 or more?


conditional = (df['number_of_reviews'] >= 100) & (df['availability_365'] == 0)
df[conditional].groupby(['host_name']).count().sort_values(['number_of_reviews'], ascending = False)[['number_of_reviews', 'availability_365']]




number_of_reviews  availability_365
host_name                                           
Michael                          5                 5
Gurpreet  Singh                  4                 4
Brian                            3                 3
Emily And Joel                   2                 2
Jo & Savannah                    2                 2
...                            ...               ...
Ehren                            1                 1
Edward                           1                 1
Dragan                           1                 1
Doug                             1                 1
正川                               1                 1

[141 rows x 2 columns]

By filtering the data, we have found that 141 hosts have met the requirements of having zero availability and at least 100 reviews.

In [119]:
# What host has the highest total of prices and where are they located?
df.groupby(['host_name']).max().sort_values(['price'], ascending = False)[['price', 'neighbourhood_group', 'neighbourhood']].head(5)


C:\Users\adoro\AppData\Local\Temp\ipykernel_13976\1100112953.py:2: FutureWarning: Dropping invalid columns in DataFrameGroupBy.max is deprecated. In a future version, a TypeError will be raised. Before calling .max, select only columns which should be valid for the function.
  df.groupby(['host_name']).max().sort_values(['price'], ascending = False)[['price', 'neighbourhood_group', 'neighbourhood']].head(5)


price neighbourhood_group neighbourhood
host_name                                         
Kathrine   10000              Queens      Kips Bay
Erin       10000              Queens  Williamsburg
Jelena     10000           Manhattan  Williamsburg
Amy         9999              Queens      Woodside
Olson       9999           Manhattan   East Harlem

The highest total of prices through our analysis are currently located in Kips Bay, Queens; Williamsburg, Queens; and Williamsburg, Manhattan.

In [128]:
# When did Danielle from Queens last receive a review?
last_review = (df['host_name'] == 'Danielle') & (df['neighbourhood_group'] == 'Queens')

df[last_review].groupby(['host_name']).max().sort_values(['neighbourhood_group'], ascending = False)['last_review']

host_name
Danielle   2019-07-08
Name: last_review, dtype: datetime64[ns]

Through all of Danielle (from Queen's) listings, her last review came on 7/8/2019.

## Further Questions

1. Which host has the most listings?

In [193]:
df.groupby(['host_name']).count().sort_values(['calculated_host_listings_count'], ascending = False)[['calculated_host_listings_count']].head(5)

calculated_host_listings_count
host_name                                   
Michael                                  417
David                                    403
Sonder (NYC)                             327
John                                     294
Alex                                     279

Based on our analysis by filtering data by calculated host listings count, Michael was noted to have the most listings with 417.

2. How many listings have completely open availability?

In [189]:
df.groupby(['availability_365']).max().sort_values(['calculated_host_listings_count'], ascending = False)[['calculated_host_listings_count']]

C:\Users\adoro\AppData\Local\Temp\ipykernel_13976\1574119878.py:1: FutureWarning: Dropping invalid columns in DataFrameGroupBy.max is deprecated. In a future version, a TypeError will be raised. Before calling .max, select only columns which should be valid for the function.
  df.groupby(['availability_365']).max().sort_values(['calculated_host_listings_count'], ascending = False)[['calculated_host_listings_count']]


calculated_host_listings_count
availability_365                                
365                                          327

Our analysis has revealed that there are 327 listings that have open availability.

3. What room_types have the highest review numbers?

In [192]:
df.groupby('room_type').max().sort_values(['number_of_reviews'], ascending = False)[['number_of_reviews', 'reviews_per_month']]

C:\Users\adoro\AppData\Local\Temp\ipykernel_13976\795695077.py:1: FutureWarning: Dropping invalid columns in DataFrameGroupBy.max is deprecated. In a future version, a TypeError will be raised. Before calling .max, select only columns which should be valid for the function.
  df.groupby('room_type').max().sort_values(['number_of_reviews'], ascending = False)[['number_of_reviews', 'reviews_per_month']]


number_of_reviews  reviews_per_month
room_type                                            
Private room                   629              58.50
Entire home/apt                488              15.78
Shared room                    454              10.77

It has been identified that the Private room has the most number of reviews as well as reviews_per_month. The next room type was the entire home/apt with 488 reviews and 15.78 per month.

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please describe them here.

-- Add your conclusion --

In [201]:
#Reviews vs. Neighbourhood Group

df.groupby('neighbourhood_group').sum().sort_values(['number_of_reviews'], ascending = False)[['number_of_reviews', 'reviews_per_month']]

C:\Users\adoro\AppData\Local\Temp\ipykernel_13976\1123081201.py:3: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby('neighbourhood_group').sum().sort_values(['number_of_reviews'], ascending = False)[['number_of_reviews', 'reviews_per_month']]


number_of_reviews  reviews_per_month
neighbourhood_group                                      
Brooklyn                        486574           21104.98
Manhattan                       454569           21158.08
Queens                          156950            8879.05
Bronx                            28371            1609.94
Staten Island                    11541             587.99

Based on my team and I's analysis, Manhattan had the most listings for Airbnb in New York, however it contains the highest average price. Brooklyn has the highest number of reviews and reviews_per_month. Therefore, if someone was thinking of coming to New York and staying at an AirBnB, I would recommend them to stay in Brooklyn in a private room as Brooklyn has a lower average price with the most amount of reviews.